winch/codegen/src/codegen/mod.rs

```diff
@@ -404,10 +408,78 @@ where
 
                 match &ty {
                     I32 | I64 | F32 | F64 => self.masm.store(src.into(), addr, ty.into()),
-                    _ => panic!("Unsupported type {:?}", ty),
+                    Ref(rt) => match rt.heap_type {
+                        WasmHeapType::Func => self.masm.store_ptr(src.into(), addr),
+                        ht => unimplemented!("Support for WasmHeapType: {ht}"),
+                    },
+                    _ => unimplemented!("Support for WasmType {ty}"),
                 }
             });
     }
+
+    /// Emits a series of instructions to lazily initialize a function reference.
+    pub fn emit_lazy_init_funcref(
+        table_data: &TableData,
+        table_index: TableIndex,
+        ptr_type: WasmType,
+        context: &mut CodeGenContext,
+        masm: &mut M,
+        call: &mut FnCall,
+        callee: Reg,
+    ) {
+        let index = context.pop_to_reg(masm, None);
+        let elem_value: Reg = context.any_gpr(masm).into();
+        let base = context.any_gpr(masm);
+        let elem_addr = masm.table_elem_address(index.into(), base, &table_data, context);
+        masm.load_ptr(elem_addr, elem_value);
+
+        let defined = masm.get_label();
+        let cont = masm.get_label();
+
+        // Preemptively move the table element address to the
+        // result register, to avoid conflicts at the control flow merge.
+        let result = call.abi_sig.result.result_reg().unwrap();
```

Comment:
Yeah, because this function should only be used to lazily initialize a funcref, we know that the builtin associated to it should always have 1 return value.